src/Compilers/CSharp/Portable/Compilation/CSharpCompilation.cs

```diff
@@ -2085,9 +2119,12 @@ public new SemanticModel GetSemanticModel(SyntaxTree syntaxTree, bool ignoreAcce
                 throw new ArgumentException(CSharpResources.SyntaxTreeNotFound, nameof(syntaxTree));
             }
 
-            return new SyntaxTreeSemanticModel(this, (SyntaxTree)syntaxTree, ignoreAccessibility);
+            return SemanticModelProvider?.GetSemanticModel(syntaxTree, this, ignoreAccessibility) ?? CreateSemanticModel(syntaxTree, ignoreAccessibility);
         }
 
+        internal override SemanticModel CreateSemanticModel(SyntaxTree syntaxTree, bool ignoreAccessibility)
```

Comment:
Should this API also perform the same argument validation as the `Get` version?

Reply:
I don't think it is required as all callers are internal and callers from public APIs would have already done the validation.